src/context/mod.rs

```diff
@@ -3,58 +3,88 @@ use std::{
     cell::{Ref, RefCell},
     ops,
     path::Path,
+    sync::Arc,
 };
 
-use derive_builder::Builder;
-use squalid::{IsEmpty, OptionExt};
-use tree_sitter_grep::{streaming_iterator::StreamingIterator, RopeOrSlice, SupportedLanguage};
+use tree_sitter_grep::{
+    streaming_iterator::StreamingIterator, tree_sitter::Tree, RopeOrSlice, SupportedLanguage,
+};
 
 mod backward_tokens;
+mod fix;
 mod get_tokens;
+mod skip_options;
 
-use get_tokens::get_tokens;
+use backward_tokens::{get_backward_tokens, get_tokens_before_node};
+pub use fix::{Fixer, PendingFix};
+use get_tokens::{get_tokens, get_tokens_after_node};
+pub use skip_options::{SkipOptions, SkipOptionsBuilder};
 
-use self::{backward_tokens::get_backward_tokens, get_tokens::get_tokens_after_node};
 use crate::{
     rule::InstantiatedRule,
     tree_sitter::{Language, Node, Query},
     violation::{Violation, ViolationWithContext},
-    Config,
+    AggregatedQueries, Config,
 };
 
-pub struct QueryMatchContext<'a> {
+#[derive(Copy, Clone)]
+pub struct FileRunContext<'a> {
```

Comment:
Split out a "shared" `FileRunContext` struct (sort of in preparation for trying to get "provided types" working which conceptually seem like they would get fed one of these to instantiate themselves against (eg the file contents + tree))